gdrive/sheets_client.py

```diff
@@ -0,0 +1,657 @@
+import logging
+import pandas as pd
+from typing import List
+
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+from gdrive import settings, error
+
+log = logging.getLogger(__name__)
+
+creds = service_account.Credentials.from_service_account_info(
+    settings.CREDENTIALS, scopes=settings.SCOPES
+)
+
+sheets_service = build("sheets", "v4", credentials=creds)
+
+"""
+At present, every function call in this library represents its own API
+transaction. If a lot of operations were to be made at once, this would hinder speed
+quite a bit. 
+
+Some batching in the future if the use case for this library gets heavier is necessary. 
+"""
+
+# Generic functions
+
+
+def update_cell_value(
+    sheet_id: str, page_name: str, range_str: str, value: str, vio="USER_ENTERED"
+):
+    """
+    Write the specifed value to specified range
+
+    Args:
+        sheet_id (str): Google sheets object ID
+        page_name (str): page target to edit
+        range_str (str): range to write the values to
+        value (str): value to write to the specified location
+        vio (str):
+            default (str): "USER_ENTERED" User entered values get resolved by googles parsing function.
+                            Functions, Integers and strings can all be entered this way.
+
+    Returns:
+        Google API Raw Result
+    """
+    body = {
+        "values": [
+            # Cell values
+            [
+                value,
+            ]
+            # Other values
+        ]
+    }
+
+    result = (
+        sheets_service.spreadsheets()
+        .values()
+        .update(
+            spreadsheetId=sheet_id,
+            range="%s!%s" % (page_name, range_str),
+            valueInputOption=vio,
+            body=body,
+        )
+        .execute()
+    )
+
+    return result
+
+
+def add_pivot_tables(
+    sheets_id: str,
+    target_page_id: str,
+    pivot_table_definition: object,
+    row_idx: int = 0,
+    col_idx: int = 0,
+):
+    """
+    Writes the pivot table definition to the specified location.
+
+    Args:
+        sheets_id (str): ID for the sheets object
+        target_page_id (str): ID for the target page of the sheets object, (Sheet1 is always 0)
+        pivot_table_definition (object): JSON encoded dict
+        row_idx (int): Index of the row to write the start of the table
+            default: 0
+        col_idx (int): Index of the column to write the start of the table
+            default: 0
+
+    Returns:
+        Google Sheets API Response: RAW response to the write operation
+    """
+    requests = []
+    requests.append(
```

Comment:
is there a reason this is written this way instaed of just setting the value of the variable?

Reply:
I dont think so. Think this is just an artifact of pulling code examples from the docs.